cronjob_scripts/stats.py

```diff
@@ -12,40 +17,65 @@
 _influxdb_client = None
 
 
-def get_stats(period: str, arch: str | None):
+@lru_cache
+def get_stats(period: str) -> DataFrame:
+    """
+    To reduce the number of round-trips to the InfluxDB server, we do a single query and cache the
+    results of this function.
+
+    In theory it would be better if we could use a dataloader-like pattern to batch up requests and
+    ask for exactly the architectures we want, but that would require us to make the whole script
+    async, and we know that cronjob will eventually ask for all architectures anyway.
+    """
     global _influxdb_client
     if _influxdb_client is None:
         _influxdb_client = InfluxDBClient3(
             host=HOST, token=TOKEN, org=ORG, database=DATABASE
         )
 
+    # Old clients (going via the old stats server) don't report status, so we filter them out.
```

Comment:
I think keeping data from old client is ok?

Reply:
yeah, fair point: phasing out the old clients seems to be taking longer than I expected, according to https://alsuren.grafana.net/d/cdx9qygf9e0owe/cargo-quickinstall?from=now-7d&to=now&timezone=browser&viewPanel=panel-3 . I will have another think about this.